From 7c959a3e938a32b386418ddfdfbb884a96762d90 Mon Sep 17 00:00:00 2001
From: Michael Orlitzky <michael@orlitzky.com>
Date: Sat, 7 Aug 2021 08:14:44 -0400
Subject: [PATCH 1/1] fflas-ffpack/interfaces/libs/Makefile.am: fix internal
 linking.

This is just,

  https://github.com/linbox-team/fflas-ffpack/pull/339

but rebased onto the old release tarball.
---
 fflas-ffpack/interfaces/libs/Makefile.am | 22 ++++++++--------------
 1 file changed, 8 insertions(+), 14 deletions(-)

diff --git a/fflas-ffpack/interfaces/libs/Makefile.am b/fflas-ffpack/interfaces/libs/Makefile.am
index 8c56746..449f6e4 100644
--- a/fflas-ffpack/interfaces/libs/Makefile.am
+++ b/fflas-ffpack/interfaces/libs/Makefile.am
@@ -24,7 +24,6 @@ pkgincludesubdir=$(pkgincludedir)/interfaces/libs
 
 AM_CXXFLAGS = @DEFAULT_CFLAGS@
 AM_CPPFLAGS = -I$(top_srcdir) #/fflas-ffpack/utils/ -I$(top_srcdir)/fflas-ffpack/fflas/  -I$(top_srcdir)/fflas-ffpack/ffpack  -I$(top_srcdir)/fflas-ffpack/field
-LDADD = $(CBLAS_LIBS) $(GIVARO_LIBS) $(CUDA_LIBS) $(PARFLAGS)
 #AM_LDFLAGS=-static 
 
 
@@ -55,31 +54,26 @@ libfflas_la_SOURCES= fflas_L1_inst.C \
 		    fflas_L3_inst.C \
 		    fflas_L3_inst_implem.inl
 
-libfflas_la_LDFLAGS=  $(LDADD) -version-info 1:0:0 \
-	             -no-undefined
+libfflas_la_LIBADD= $(CBLAS_LIBS) $(GIVARO_LIBS) $(CUDA_LIBS) $(PARLIBS)
+libfflas_la_LDFLAGS= -version-info 1:0:0 -no-undefined
 
 libffpack_la_SOURCES= ffpack_inst.C \
 		      ffpack_inst_implem.inl
-libffpack_la_LDFLAGS= $(LDADD) -version-info 1:0:0 \
-		       -no-undefined $(top_builddir)/fflas-ffpack/interfaces/libs/libfflas.la
-
-EXTRA_libffpack_la_DEPENDENCIES= libfflas.la
+libffpack_la_LIBADD= libfflas.la
+libffpack_la_LDFLAGS= -version-info 1:0:0 -no-undefined
 
 libfflas_c_la_SOURCES=fflas_lvl1.C \
 		    fflas_lvl2.C \
 		    fflas_lvl3.C \
 		    fflas_sparse.C
 #libfflas_c_la_CPPFLAGS=$(AM_CPPFLAGS) -DFFLAS_COMPILED -DFFPACK_COMPILED
-libfflas_c_la_LDFLAGS=  $(LDADD) -version-info 1:0:0 \
-                       -no-undefined 	$(top_builddir)/fflas-ffpack/interfaces/libs/libfflas.la
-
-EXTRA_libfflas_c_la_DEPENDENCIES=libfflas.la
+libfflas_c_la_LIBADD= libfflas.la
+libfflas_c_la_LDFLAGS= -version-info 1:0:0 -no-undefined
 
 libffpack_c_la_SOURCES=ffpack.C
 #libffpack_c_la_CPPFLAGS=$(AM_CPPFLAGS) -DFFLAS_COMPILED -DFFPACK_COMPILED
-libffpack_c_la_LDFLAGS=  $(LDADD) -version-info 1:0:0 \
-		        -no-undefined -lfflas -lffpack
-EXTRA_libffpack_c_la_DEPENDENCIES=libffpack.la
+libffpack_c_la_LIBADD= libffpack.la
+libffpack_c_la_LDFLAGS= -version-info 1:0:0 -no-undefined
 
 
 EXTRA_DIST=c_libs.doxy
-- 
2.31.1

